### PR TITLE
Update dependencies, and use golangci-lint

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

Depends on https://github.com/signalfx/sapm-proto/pull/21

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# See https://github.com/golangci/golangci-lint#config-file
+run:
+  issues-exit-code: 1 #Default
+  tests: true #Default
+
+linters:
+  enable:
```
Comment (jgheewala) [inline .golangci.yml:7]: is there plan to enable more linters? Also any reason for just this linters?

**Review by jgheewala — APPROVED** [review]

**Comment by jgheewala** [comment]
seems like circle ci build failed with "staticcheck"

**Review by jgheewala — CHANGES_REQUESTED** [review]
linting failures

File: .golangci.yml (hunk 1 above)
Comment (bogdandrutu) [reply .golangci.yml:7]: This is on top of the default linters.

**Review by jgheewala — APPROVED** [review]
lgtm

**Pull request merged**